composer.json

```diff
@@ -28,6 +28,9 @@
         "openlss/lib-array2xml": "^1.0",
         "symfony/expression-language": "^2.3|^3.0|^4.0|^5.0|^6.0"
     },
+    "conflict": {
+        "phpunit/phpunit": ">=11.0"
+    },
```

Comment:
Actually `"phpunit/phpunit": "~9 || ~10"` in tools should work even better but I'm starting to think that we might want to move PHPUnit into main composer.json require-dev section since phpmatcher is proving an integration with it. If PHPUnit would be added to composer.json instead of tools, test workflows would detect that since they should be expected at lowest/current/latest dependencies.

Reply:
> Modifying tools / seems more natural.

The `require-dev` part is not keep in mind for project using the php-matcher and so will fail like in #463 and we recommend to this and not just set PHPUnit now to ^10 because else all will get failing version 6.0.12 on phpunit 9, so at current case I recommend to release a PHPUnit 9 compatible version, before requiring only PHPUnit 10.

I did go with `conflict` this is the only way of handling `optionalDependencies` just added it to avoid that incompatible version is installed in future. The only way would be move it as hard dependencies. Which is fine from my side, but I understand that maybe somebody don't want that PHPUnit is installed when using the package, but in my case fine to move it to `require`.

Reply:
Lets move phpunit to require-dev in main composer.json - this will not make composer to install PHPUnit for everyone who is using phpmatcher, but it will let test suite to run against different PHPUnit versions. 

https://phpunit.de/supported-versions.html

According to this page I think matcher 6.x should support `~8 || ~9 || ~10` versions of PHPUnit.

Reply:
We still require the `conflict` part as described above the `require-dev` does not disallow to use php-matcher with an incompatible version.

Reply:
yeah, that is indeed weird, could you maybe try to remove composer.lock and run `composer update`? If that won't work I will take a look at what's going on there.

Reply:
which php version I should use to update the `composer.lock`?

Reply:
8.1 should be fine as its the lowest supported version

Reply:
I think now all should be fine.